test/e2e/performanceprofile/functests/utils/tuned/tuned.go

```diff
@@ -84,38 +84,34 @@ func GetPod(ctx context.Context, node *corev1.Node) (*corev1.Pod, error) {
 }
 
 func WaitForStalldTo(run bool, interval, timeout time.Duration, node *corev1.Node) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
-		cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
-		stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
-		if err != nil {
-			return false, fmt.Errorf("failed to execute command %q on node: %q; %w", cmd, node.Name, err)
-		}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		cmd := []string{"/bin/bash", "-c", "pidof stalld"}
```

Comment:
When stalld disabled, `pidof stalld` will return with an error, this is why I added the `|| true`.
Plus not checking the error returned from `ExecCommandOnNode` is also not so great.

Reply:
there were some test cases where the the command itself with the condition actually failed. this is why we moved to a different kind of check (see the rest of the code)

Reply:
The problem is that we're not checking the error returned from `nodes.ExecCommandOnNode(cmd, node) ` hence we don't know whether the command execution was successful.

I don't mind changing the executed command  to something that doesn't contain pipes but we need to check the error anyway